gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -362,6 +367,39 @@ static void checkValidFloatingPoint(double value) {
     }
   }
 
+  private static TypeAdapterFactory objectAdapterFactory(ToNumberStrategy objectToNumberStrategy) {
+    if (objectToNumberStrategy == ToNumberPolicy.DOUBLE) {
+      return ObjectTypeAdapter.FACTORY;
+    }
+    return ObjectTypeAdapter.newFactory(objectToNumberStrategy);
+  }
+
+  private static TypeAdapterFactory numberAdapterFactory(final ToNumberStrategy numberToNumberStrategy) {
+    if (numberToNumberStrategy == ToNumberPolicy.LAZILY_PARSED_NUMBER) {
+      return TypeAdapters.NUMBER_FACTORY;
```

Comment:
Maybe prevent code duplication by removing `TypeAdapters.NUMBER_FACTORY` and instead create a class similar to `ObjectTypeAdapter` which then contains the constant and the logic of this `numberAdapterFactory` method